### ISSUE TITLE
Language detect fails on successive visits

### ISSUE THREAD

**Issue by danielguillan** [getkirby/kirby#268]
When enabling `language.detect` on a multilingual website the redirect works as expected when a user with no cookies visits a page for the first time. On a second visit the user is redirected to the site's default language instead of the detected one.

Example config:

```php
c::set('languages', array(
  array(
    'default' => true,
    'code'    => 'es',
    'name'    => 'Español',
    'locale'  => 'es_ES',
    'url'     => '/es',
  ),
  array(
    'code'    => 'en',
    'name'    => 'English',
    'locale'  => 'en_US',
    'url'     => '/en',
  ),
  array(
    'code'    => 'fr',
    'name'    => 'Français',
    'locale'  => 'fr_FR',
    'url'     => '/fr',
  )
));

c::set('language.detect', true);
`````
- User browser language set to English.
- First visit to `/`, user is correctly redirected to `/en`.
- Second visit to `/`, user is redirected to the default language (`/es`) instead of `/en`.

**Comment by iamvdo** [comment]
Same issue here. Any workaround ?

**Comment by lukasbestle** [comment]
Does this also happen for users that aren't logged in to the Panel?

Also, @bastianallgeier: Why is the default language being used in [kirby.php:309](https://github.com/getkirby/kirby/blob/master/kirby.php#L309) instead of using the stored language that is returned by `Site->sessionLanguage()`?

**Comment by danielguillan** [comment]
> Does this also happen for users that aren't logged in to the Panel?

Yes.

**Comment by mauricerenck** [comment]
I don't know if this helps, here's what I get as a result, when for example viewing a category of my blog:

![default](https://cloud.githubusercontent.com/assets/964547/10453792/95049c3a-71af-11e5-9747-cf3467df8238.png)

The URL-language comes from my route (here is the code, in case I made a mistake there): 

    array(
        'pattern' => '([a-z]{2})/blog/category/(:any)',
        'action'  => function($lang, $tag) {

            // make sure that the blog page is 
            // being set as the currently active page
            site()->visit('blog');

            // send the tag to the template
            return array('blog', array(
                'lang' => $lang,
                'tag' => $tag
            ));

        }
    )

**Comment by lukasbestle** [comment]
You should be able to use the following code:

```php
array(
    'pattern' => '([a-z]{2})/blog/category/(:any)',
    'action'  => function($lang, $tag) {

        // make sure that the blog page is 
        // being set as the currently active page
        site()->visit('blog', $lang);

        // send the tag to the template
        return array('blog', array(
            'tag' => $tag
        ));

    }
)
```

**Comment by bastianallgeier** [comment]
I finally found a fix for this. So sorry that it took so long. The fix will be included in 2.2
